frontend/src/components/object/share/ShareObjectButton.vue

```diff
@@ -0,0 +1,108 @@
+<script setup lang="ts">
+import { computed, ref, type PropType } from 'vue';
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
+import { storeToRefs } from 'pinia';
+
+import Button from 'primevue/button';
+import Dialog from 'primevue/dialog';
+import TabView from 'primevue/tabview';
+import TabPanel from 'primevue/tabpanel';
+
+import ShareLinkContent from './ShareLinkContent.vue';
+import { useConfigStore } from '@/store';
+
+import type { COMSObject } from '@/interfaces';
+
+const { config } = storeToRefs(useConfigStore());
+
+// Props
+const props = defineProps({
+  obj: {
+    type: Object as PropType<COMSObject>,
+    required: true,
+  },
+});
```

Comment:
Can we do a pure type annotation here instead of the extra wrapping?

const props = defineProps<{
  obj: COMSObject
}>();